### PR TITLE
EWPP-3315: Remove projecy budget fallback logic.

### PR DESCRIPTION
**Pull request by nagyad**

### CODE REVIEW

File: modules/oe_theme_content_project/src/Plugin/ExtraField/Display/OverallBudgetExtraField.php [hunk 1]
```diff
@@ -74,7 +74,7 @@ public function getLabel() {
    * {@inheritdoc}
    */
   public function viewElements(ContentEntityInterface $entity) {
-    if ($entity->get('oe_project_budget')->isEmpty() && $entity->get('oe_project_eu_budget')->isEmpty()) {
+    if ($entity->get('oe_project_eu_budget')->isEmpty()) {
```
Comment (22Alexandra) [inline modules/oe_theme_content_project/src/Plugin/ExtraField/Display/OverallBudgetExtraField.php:77]: We should not return empty arrays but use the isEmpty property.

File: modules/oe_theme_content_project/src/Plugin/ExtraField/Display/PercentageExtraField.php [hunk 2]
```diff
@@ -89,24 +89,19 @@ public function viewElements(ContentEntityInterface $entity) {
...
     // Return only EU contribution if budget is empty.
-    if ($entity->get('oe_project_budget')->isEmpty() && $entity->get('oe_project_eu_budget')->isEmpty()) {
+    if ($entity->get('oe_project_eu_budget')->isEmpty()) {
       return $build;
     }
 
     // Compute budget percentage field value.
-    $budget = $entity->get('oe_project_eu_budget')->isEmpty() ? $entity->get('oe_project_budget_eu')->value : $entity->get('oe_project_eu_budget')->value;
+    $budget = $entity->get('oe_project_eu_budget')->value;
+    $budget_eu = $entity->get('oe_project_eu_contrib')->value;
```
Comment (22Alexandra) [inline modules/oe_theme_content_project/src/Plugin/ExtraField/Display/PercentageExtraField.php]: We're still rendering 0% contribution which I don't think is needed.

**Review by 22Alexandra — CHANGES_REQUESTED** [review]

File: modules/oe_theme_content_project/src/Plugin/ExtraField/Display/OverallBudgetExtraField.php (hunk 1 above)
Comment (22Alexandra) [reply modules/oe_theme_content_project/src/Plugin/ExtraField/Display/OverallBudgetExtraField.php:77]: Ok this is good now but we still need to `return;` when there is nothing else to do.

File: modules/oe_theme_content_project/src/Plugin/ExtraField/Display/PercentageExtraField.php [hunk 3]
```diff
@@ -74,8 +74,8 @@ public function getLabel() {
    * {@inheritdoc}
    */
   public function viewElements(ContentEntityInterface $entity) {
-    if ($entity->get('oe_project_budget_eu')->isEmpty() && $entity->get('oe_project_eu_contrib')->isEmpty()) {
-      return [];
+    if ($entity->get('oe_project_eu_budget')->isEmpty() && $entity->get('oe_project_eu_contrib')->isEmpty()) {
+      $this->isEmpty = TRUE;
```
Comment (22Alexandra) [inline modules/oe_theme_content_project/src/Plugin/ExtraField/Display/PercentageExtraField.php:78]: Same as above.

**Review by 22Alexandra — CHANGES_REQUESTED** [review]

File: modules/oe_theme_content_project/src/Plugin/ExtraField/Display/OverallBudgetExtraField.php (hunk 1 above)
Comment (22Alexandra) [reply modules/oe_theme_content_project/src/Plugin/ExtraField/Display/OverallBudgetExtraField.php:77]: Nvm.

**Review by 22Alexandra — DISMISSED** [review]

File: modules/oe_theme_content_project/src/Plugin/ExtraField/Display/PercentageExtraField.php [hunk 4]
```diff
@@ -89,28 +89,23 @@ public function viewElements(ContentEntityInterface $entity) {
...
       return $build;
     }
 
-    // Compute budget percentage field value.
-    $budget = $entity->get('oe_project_eu_budget')->isEmpty() ? $entity->get('oe_project_budget_eu')->value : $entity->get('oe_project_eu_budget')->value;
-    $percentage = $this->getPercentage((float) $budget, (float) $budget_eu);
-    $build[] = [
-      '#markup' => '<div class="ecl-u-mt-m">' . $this->t("@percentage% of the overall budget", ["@percentage" => $percentage]) . '</div>',
-    ];
+    if (!$entity->get('oe_project_eu_contrib')->isEmpty() && !$entity->get('oe_project_eu_contrib')->isEmpty()) {
```
Comment (sergepavle) [inline modules/oe_theme_content_project/src/Plugin/ExtraField/Display/PercentageExtraField.php]: Repeated condition

**Review by sergepavle — CHANGES_REQUESTED** [review]

**Review by sergepavle — APPROVED** [review]

**Review by 22Alexandra — APPROVED** [review]

**Review by 22Alexandra — APPROVED** [review]

**Pull request merged**